### PR TITLE
MAIN -  adds accessibility features  on the modal

### PR DESCRIPTION
**Pull request by paschalidi**
👋 to you John! 
As we talked in person I made this list of all the accessible 'features' a modal should include. You could find the list yourself [here](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal).

What is already there.
✅ When the Modal is not open, it is not rendered into the DOM.
✅ Clicking outside the Modal will close it.
✅ When open, scrolling is frozen on the main document beneath the Modal.

**What this PR tackles.**
✅  The Modal has relevant WAI-ARIA attributes in accordance with accessibility guidelines.
✅  When open, focus is drawn immediately to the Modal's close button.
✅  Focus is trapped within the Modal when open.

**What is to come 🥁🥁**
1️⃣  When the Modal closes, focus returns to the Modal's trigger button.
2️⃣  Pressing the escape key will close the Modal.

### CODE REVIEW

**Comment by paschalidi** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Missing reviewer - <i>Looks like this pull request is missing a reviewer. That's OK as long as this PR is still a 'work in progress', if that is the case please add 'WIP' in the PR title or corresponding label.</i></td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>✅  Get a 🏆 for adding description on this PR. - <i>Celebrate on every step!</i></td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c183c0cf5adda2135993654990b1f6b99a412296
</p>

File: src/Modal/Modal.jsx [hunk 1]
```diff
@@ -13,12 +23,29 @@ class Modal extends Component {
   }
 
   render() {
-    const { onClose, children } = this.props;
+    const { closeCallback, role, ariaLabel, children } = this.props;
```
Comment (Weetbix) [inline src/Modal/Modal.jsx]: This has the old `closeCallback` method name

**Comment by paschalidi** [comment]
@Weetbix have another look when you get the chance please :D Sorry for not taking good care of this at the first place. My eyes are bad.

File: src/Modal/Modal.jsx [hunk 2]
```diff
@@ -13,12 +23,29 @@ class Modal extends Component {
...
+          <Overlay onClick={onClose} />
+          <Content>
+            <Row textAlign="right">
+              <Col offset={11} lg={1}>
+                <CloseButton ref={this.onClose} onClick={onClose}>
+                  <Icon name="close" />
+                </CloseButton>
+              </Col>
+            </Row>
+            <div style={{ marginRight: 24 }}>{children}</div>
```
Comment (Weetbix) [inline src/Modal/Modal.jsx]: I thought we will keep all the styling in the styles files? 

Also the left margin should be the same: 
image

**Comment by paschalidi** [comment]
@Weetbix fixed!

**Comment by paschalidi** [comment]
@Weetbix fixed it :| sorry man

**Pull request merged**